### PR TITLE
Allow crux-sudo-edit on remote files

### PR DESCRIPTION
**Pull request by dieggsy**
- If editing with sudo method or similar alternative, or user is 'root',
  do nothing.
- If on localhost, use sudo method and switch to root user - if on
  remote host, switch to root user.

- Restore place.
- Remove ido dependency.

On ido: just felt kind of weird to be hard coding ido-read-file-name in here, esp. given that it's a preference thing, a lot of people these days don't use ido, and current alternatives (helm, ivy) wrap/advise read-file-name anyway.

### CODE REVIEW

File: crux.el [hunk 1]
```diff
@@ -456,23 +456,47 @@ See `file-attributes' for more info."
...
-(require 'ido)
 ;;;###autoload
 (defun crux-sudo-edit (&optional arg)
   "Edit currently visited file as root.
 
 With a prefix ARG prompt for a file to visit.
 Will also prompt for a file to visit if current
 buffer is not visiting a file."
-  (interactive "P")
+  (interactive)
```
Comment (bbatsov) [inline crux.el]: Seems to me you made a mistake removing this.

File: crux.el [hunk 2]
```diff
@@ -456,23 +456,47 @@ See `file-attributes' for more info."
   (equal (crux-file-owner-uid filename)
          (user-uid)))
 
+(defun crux-already-root-p ()
+  (let ((remote-method (file-remote-p default-directory 'method))
+        (remote-user (file-remote-p default-directory 'user)))
+    (when (and remote-method
+               (or (member remote-method '("sudo" "su" "ksu" "doas"))
+                   (string= remote-user "root")))
+      (message "Already editing this file as root."))))
```
Comment (bbatsov) [inline crux.el]: I don't think that a predicate method should have side effects (like printing a message).

File: crux.el (hunk 1 above)
Comment (dieggsy) [reply crux.el]: Whoops! Agreed.

File: crux.el (hunk 2 above)
Comment (dieggsy) [reply crux.el]: Agreed.

**Comment by dieggsy** [comment]
Made the necessary changes according to your suggestions, thanks!

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**